lib/seahorn/BvOpSem2Allocators.cc

```diff
@@ -296,7 +301,15 @@ class StaticOpSemAllocator : public OpSemAllocator {
       preAlloc(inst, memSz, true);
     } else {
       // -- allocate max allowed for dynamically sized allocations
-      preAlloc(inst, m_maxSymbAllocSz, false);
+      // TODO: use rng info here
+      auto operand_inst = dyn_cast<llvm::Instruction>(inst.getOperand(0));
+      auto rangeVal = m_sem.getInstRng(*operand_inst);
+      unsigned memSz =
+          rangeVal.second != 0 ? rangeVal.second : m_maxSymbAllocSz;
```

Comment:
this looks wrong. It is possible to have `malloc(0)`. So `0` is a reasonable upper bound for the range. Only when upper bound is unbounded we need to fall back to `maxSymbAllocSz`